qt/applications/workbench/workbench/app/mainwindow.py

```diff
@@ -416,21 +425,27 @@ def add_dockwidget(self, plugin):
 
     # ----------------------- Layout ---------------------------------
 
-    def setup_layout(self):
-        """Assume this is a first run of the application and set layouts
-        accordingly"""
-        self.setup_default_layouts()
+    def save_layout_settings(self):
+        """Save the current layout to configuration"""
+        CONF.set("Layout/Layout", self.saveState())
+
+    def load_layout_settings(self):
+        """Load a layout for the child widgets from the user config, if no user default exists load the default"""
+        try:
+            self.restoreState(CONF.get("Layout/Layout"))
```

Comment:
I know it was my suggestion to create a `[Layout]` section, but looking at the .ini file there's an existing `[MainWindow]` section. Perhaps it makes sense to save this to `MainWindow/UserLayout` or something like that?

Reply:
```suggestion
            self.restoreState(CONF.get("MainWindow/UserLayout"))
```